src/App.js

```diff
@@ -1,24 +1,74 @@
-import logo from './logo.svg';
-import './App.css';
+import Navbar from "./Components/Navbar";
+import { useEffect, useState } from "react";
+import { fetchFolderData, fetchUserData } from "./api";
+import Profile from "./Components/Profile";
+import CardList from "./Components/CardList";
+import Footer from "./Components/Footer";
+import { Route, Routes } from "react-router-dom";
+import Detail from "./Components/Detail";
 
 function App() {
+  const [user, setUser] = useState({
+    id: null,
+    name: null,
+    email: null,
+    profileImageSource: null,
+  });
+  const [folderData, setFolderData] = useState(null);
+
+  const loadUserData = async () => {
+    let result;
+
+    try {
+      result = await fetchUserData();
+    } catch (error) {
+      console.error(error);
+      return;
+    }
+
+    const { id, name, email, profileImageSource } = result;
+
+    setUser((p) => ({
+        ...p,
+        id: id,
+        name: name,
+        email: email,
+        profileImageSource: profileImageSource,
+    }));
+  }
```

Comment:
@HMRyu 

함수가 처리하고 있는 역할이 너무 큽니다! 
1. 데이터 조회하고 
2. 에러핸들링 하구
3. 상태 업데이트 해주는
이 모든 역할을 한 함수에 구현되어있는데요. 

이 보다는, useEffect 내에서 데이터를 불러와 에러핸들링 및 상태 업데이트를 해주는게 조금 더 좋을 것 같아요.

Reply:
또한 좀 번복적인 형태로 비동기 처리를 하고 있는데요, 이런 형태로 작성되는게 더 좋을 것 같아요!
```suggestion
  const loadUserData = async () => {
    try {
      const result = await fetchUserData();
      const { id, name, email, profileImageSource } = result;
      setUser((p) => ({
        ...p,
        id: id,
        name: name,
        email: email,
        profileImageSource: profileImageSource,
      }));
    } catch (error) {
      console.error(error);
    }
  }
```